File: client/src/androidTest/java/com/jraska/github/client/FakeFirebaseModule.java
package com.jraska.github.client;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.jraska.github.client.analytics.AnalyticsProperty;
import com.jraska.github.client.analytics.EventAnalytics;
import com.jraska.github.client.logging.CrashReporter;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class FakeFirebaseModule extends FirebaseModule {
  @Override EventAnalytics eventAnalytics(FirebaseEventAnalytics analytics) {
    return EventAnalytics.EMPTY;
  }

  @Override AnalyticsProperty analyticsProperty(FirebaseEventAnalytics analytics) {
    return mock(AnalyticsProperty.class);
  }

  @Override CrashReporter firebaseCrash() {
    return mock(CrashReporter.class);
  }

  @Override FirebaseDatabase firebaseDatabase() {
    FirebaseDatabase mockedDatabase = mock(FirebaseDatabase.class);
    when(mockedDatabase.getReference(any())).thenReturn(mock(DatabaseReference.class));
    return mockedDatabase;
  }
}

File: client/src/androidTest/java/com/jraska/github/client/FakeFirebaseModule.kt
package com.jraska.github.client

import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.jraska.github.client.analytics.AnalyticsProperty
import com.jraska.github.client.analytics.EventAnalytics
import com.jraska.github.client.logging.CrashReporter
import org.mockito.ArgumentMatchers.any
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

class FakeFirebaseModule : FirebaseModule() {
  override fun eventAnalytics(analytics: FirebaseEventAnalytics): EventAnalytics {
    return EventAnalytics.EMPTY
  }

  override fun analyticsProperty(analytics: FirebaseEventAnalytics): AnalyticsProperty {
    return mock(AnalyticsProperty::class.java)
  }

  override fun firebaseCrash(): CrashReporter {
    return mock(CrashReporter::class.java)
  }

  override fun firebaseDatabase(): FirebaseDatabase {
    val mockedDatabase = mock(FirebaseDatabase::class.java)
    `when`(mockedDatabase.getReference(any())).thenReturn(mock(DatabaseReference::class.java))
    return mockedDatabase
  }
}
